### DISCUSSION TITLE
directus SDK / Typescript project

### DISCUSSION THREAD

**Discussion by DeanKamali** [directus/directus#8544] [Archive]
I'm following [the guide on how to use the SDK](https://docs.directus.io/reference/sdk/#typescript) in my project, and I'm getting this error when I'm starting my node server. 

Here is my code: I have generated type definitions using this `https://www.npmjs.com/package/directus-typescript-gen`. 

```
import { Directus } from '@directus/sdk';
import { MyCollections } from "./my-collections.d.ts";
const instance = new Directus<MyCollections>('http://localhost:8055/');
```

Here are the errors I'm getting: 

```

node_modules/@directus/sdk/dist/items.d.ts:36:30 - error TS1005: ']' expected.

36     [K in keyof QueryMany<T> as `_${string & K}`]: QueryMany<T>[K];
                                ~~

node_modules/@directus/sdk/dist/items.d.ts:36:49 - error TS1005: ';' expected.

36     [K in keyof QueryMany<T> as `_${string & K}`]: QueryMany<T>[K];
                                                   ~

node_modules/@directus/sdk/dist/items.d.ts:36:50 - error TS1128: Declaration or statement expected.

36     [K in keyof QueryMany<T> as `_${string & K}`]: QueryMany<T>[K];
                                                    ~

node_modules/@directus/sdk/dist/items.d.ts:37:1 - error TS1128: Declaration or statement expected.

37 };
   ~
```

I'm using the latest version of directus and the SDK

**Comment by rijkvanzanten** [comment]
What version of TypeScript are you using? Also, FWIW, we didn't create nor maintain that package, so I can't do much if the problem exists in usage of that package specifically

**Reply by DeanKamali** [reply]
Thanks for your quick reply, updating the typescript version solved the issue! 

thanks again

**Reply by joselcvarela** [reply]
Also, we may include this feature in core in a near future. See https://github.com/directus/directus/pull/8531

**Reply by DeanKamali** [reply]
It would be great actually as more people are adopting Typescript.
